src/lib/getAwsAccountFromString.ts

```diff
@@ -0,0 +1,54 @@
+import { getAlksAccounts } from './getAlksAccounts';
+import { AwsAccount } from '../model/awsAccount';
+import ALKS from 'alks.js';
+
+const accountIdRegex = /^[0-9]{12}/;
+// This alias regex was sourced from AWS's docs here -> https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateAccountAlias.html
+const aliasRegex = /^[a-z0-9](([a-z0-9]|-(?!-))*[a-z0-9])?$/;
+
+/**
+ * Gets an ALKS Account object from a user-provided account string. The user must have access to the account for it to be resolved
+ *
+ * @param accountString - a string uniquely identifying an AWS account that can take many forms, such as "012345678910/ALKSAdmin - awstest123", "012345678910/ALKSAdmin", "awstest123", or "012345678910"
+ */
+export async function getAwsAccountFromString(
+  accountString: string
+): Promise<AwsAccount | undefined> {
+  const accounts = await getAlksAccounts();
+
+  const accountId = (accountString.match(accountIdRegex) ?? [undefined])[0];
+  const alias = (accountString.match(aliasRegex) ?? [undefined])[0];
+
+  let alksAccount: ALKS.Account | undefined;
+  if (accountId) {
+    // Get a list of accounts whose account string (e.g. "012345678910/ALKSAdmin - awstest123") starts with an account ID that matches the provided account ID
+    const matchingAccounts = accounts.filter(
+      (account) => accountId === account.account.substring(0, 12)
+    );
+
+    if (matchingAccounts.length > 0) {
+      alksAccount = matchingAccounts[0];
+    }
+  } else if (alias) {
+    // Get a list of accounts whose account string (e.g. "012345678910/ALKSAdmin - awstest123") contains an alias that matches the provided alias
+    const matchingAccounts = accounts.filter(
+      (account) =>
+        account.account.match(new RegExp(` - ${alias}$`))?.length === 1
+    );
+
+    if (matchingAccounts.length > 0) {
+      alksAccount = matchingAccounts[0];
+    }
+  }
+
+  if (alksAccount) {
+    return {
+      id: alksAccount.account.substring(0, 12),
+      alias: alksAccount.account.split(' - ')[1],
+      label: alksAccount.skypieaAccount?.label,
+    };
+  }
+
+  // if no matches or too many matches were found
```

Comment:
I dont see here where you are confirming that there was only one match.  Looks like you just took the first match regardless of how many there were.  Either that logic should change or this comment should change to reflect that, unless I missed something.

Reply:
Oh sorry that was from an early attempt at this. It's actually not a problem if multiple matches are found since "ALKS Account" actually means a pairing of AWS Account and IAM Role so it's common to have multiple ALKS Accounts for the same AWS Account. For example, I have the Admin and Security role for ALKS - Non Prod, so that would mean that if I passed `awsalksnp` as the alias then this would find 2 matches: `336692926357/ALKSAdmin` and `336692926357/ALKSSecurity`. 

Basically I realized that multiple matches are ok but forgot to remove this comment